스타일 컴포넌트와 에디터 병합

ImuruKevol opened the pull request:
- 원 라인 마크다운 룰과 스타일 컴포넌트를 에디터와 병합
- 테마 컬러 정의
- 글로벌하게 쓰는 사용자 정의 타입을 enums 디렉토리에 생성
- 기능과 내용에 맞지 않는 파일명, 변수명, 위치 수정

Commit:
Refactor: EditorInput 병합 및 커서 위치 버그 수정

EditorInput을 MarkdownRenderer에 병합
div의 ContentEditable에서 커서 위치가 설정되지 않는 현상을 수정

Commit:
Fix: 텍스트 입력시 스토어에 저장되는 구조 변경

기존: 텍스트 입력시마다 디스패치
변경: 포커스 이동시마다 디스패치

Commit:
Feat: 테마 컬러 정의

테마 컬러 정의

Commit:
Chore: 셀의 태그에 대한 enumeration을 정의

[사용자 정의 타입 이름]: [실제 html tag]
의 형태로 정의

Commit:
Feat: 마크다운 문법에 따른 셀 변환 기능 추가

마크다운 문법에 따라 셀 변환 기능 추가
- 텍스트가 있는 상태에서는 아직 불가능

Commit:
Refactor: 기능과 맞지 않는 이름을 가진 파일명과 변수명 수정

기능과 맞지 않는 이름을 가진 파일명과 변수명 수정
EditorInput의 Object들을 다른 enums로 이동

Commit:
Feat: CellStore, MarkdownCell에 prop-types 적용

CellStore, MarkdownCell에 prop-types 적용

RBJH commented on client/src/actions/CellAction.js:
```diff
@@ -28,6 +30,7 @@ const cellActionCreator = {
       type: CELL_ACTION.INIT,
       renderTarget,
       index: index || 0,
+      tag: CELL_TYPE.DEFAULT,
```
CELL_TAG로 이름을 통일하는 것이 혼란이 되지 않을 것 같아요!

ImuruKevol replied:
혼란이 좀 있겠네요. 고치겠습니다.

dimsssss commented on client/src/components/editor/MarkdownCell.jsx:
```diff
@@ -23,36 +36,35 @@ const MarkdownTransformer = ({ cellUuid }) => {
+
+      const front = text.slice(0, cursor.start);
+      const back = text.slice(cursor.start, text.length);
+      const content = `${front}<span id="cursorCaret"></span>${back}`;
```
어떤 것의 앞이고 뒤 인지 구체적이면 더 좋을 것 같습니다.

ImuruKevol replied:
맨 처음에 커서를 주기 위해 임시로 만든 변수라서 더 자세히 쓰지는 않았는데 좀 더 명확하게 쓰겠습니다 :)

dimsssss commented on client/src/components/editor/MarkdownCell.jsx:
```diff
@@ -129,7 +141,7 @@ const MarkdownTransformer = ({ cellUuid }) => {
     //         cellActionCreator.init(
     //           (index, ref) => (
-    //             <MarkdownTransformer cellIndex={index} inputRef={ref} />
+    //             <MarkdownCell cellIndex={index} inputRef={ref} />
     //           ),
     //           currentIndex
     //         )
```
안 쓰는 주석을 삭제하면 더 보기 좋을 것 같습니다

ImuruKevol replied:
저번 PR에서도 말씀드렸다시피 원래는 들어가야할 기능인데 다른 버그때문에 임시로 비활성화해놓은 기능이라 일부러 안뺐습니다.
130, 131 라인에 설명이 있습니다.

dimsssss reviewed (commented):
고생하셨습니다!

Commit:
Fix: PR 코멘트 반영

PR 코멘트 반영
- MarkdownCell의 front/back 변수 네이밍 수정
- CELL_TYPE에서 CELL_TAG로 네이밍 일괄 수정

PR #122
